client/blocks/upload-image/index.jsx

```diff
@@ -300,7 +315,31 @@ class UploadImage extends Component {
 		if ( uploadedImage && uploadedImage.URL ) {
 			return (
 				<div className="upload-image__uploading-done-container">
-					<img src={ uploadedImage.URL } />
+					<Button
+						onClick={ this.removeUploadedImage }
+						compact
+						className="upload-image__uploaded-image-remove"
+					>
+						<Gridicon
+							icon="cross"
+							size={ 24 }
+							className="upload-image__uploaded-image-remove-icon"
+						/>
+					</Button>
```

Comment:
If this button was moved in DOM _after_ the `upload-image__uploaded-image-wrapper` element, you wouldn't need to set a special `z-index` for it -- just like you don't need to for the 'edit' button.